src/resolver/Resolver.js

```diff
@@ -196,8 +196,9 @@ export default class Resolver extends React.Component {
 
     // Both those props provided by parent & dynamically resolved
     return this.props.children({
-      ...this.props.props,
       ...this.state.resolved,
+      ...this.props.props,
+      props: this.props.props
```

Comment:
If you could change this to `...this.props.props` i will merge it.

I did some tests and this should not have any consequences.